src/domain/hooks/helpers/event-cache.helper.ts

```diff
@@ -508,8 +520,41 @@ export class EventCacheHelper {
     return [this.safeRepository.clearIsSafe(event)];
   }
 
+  private onTransactionEventDelegate(
+    event: Extract<
+      Event,
+      {
+        type:
+          | TransactionEventType.NEW_DELEGATE
+          | TransactionEventType.UPDATED_DELEGATE
+          | TransactionEventType.DELETED_DELEGATE;
+      }
+    >,
+  ): Array<Promise<void>> {
+    if (!event.safeAddress) {
+      return [];
```

Comment:
I'm not entirely sure why or how this can happening but the event can potentially have no `address` according to the [Transaction Service](https://github.com/safe-global/safe-transaction-service/blob/main/safe_transaction_service/history/services/notification_service.py#L237).

Reply:
Just to comment. Right now, it is possible to create a delegate for a delegator without a specific safe. In that case, the delegate user would be valid for all safes of the delegator user. If a specific safe is specified, it would only be valid for operations on that safe.

Reply:
Thanks guys! Then if I understood correctly, if the Safe `address` included in the event is `null` we would need to clear the cache for _all the Safe addresses the delegator owns_. Do you think this is correct? @iamacook @falvaradorodriguez? 🙂

Reply:
I'm merging this one as it is, and I'm opening a new ticket so we can handle that case separately: https://github.com/safe-global/safe-client-gateway/issues/2146